pyccl/halos/concentration/bhattacharya13.py

```diff
@@ -13,41 +14,26 @@ class ConcentrationBhattacharya13(Concentration):
     By default it will be initialized for Delta = 200-critical.
 
     Args:
-        mdef (:class:`~pyccl.halos.massdef.MassDef`): a mass
+        mass_def (:class:`~pyccl.halos.massdef.MassDef`): a mass
             definition object that fixes
             the mass definition used by this c(M)
             parametrization.
     """
     name = 'Bhattacharya13'
 
-    def __init__(self, mdef=None):
-        super(ConcentrationBhattacharya13, self).__init__(mdef)
+    @warn_api(pairs=[("mdef", "mass_def")])
+    def __init__(self, *, mass_def=MassDef(200, 'critical')):
```

Comment:
Won't this be running a lot of unnecessary code as soon as we import pyccl?

Reply:
It is super-super fast, and even if it was slow, it would be better to run it once at the beginning, that having to re-initialize one every time you create a `Concentration` object.
![Screenshot from 2023-04-01 04-40-12](https://user-images.githubusercontent.com/22447385/229259650-713675ee-5649-4a3e-a537-be934038b8ae.png)

Reply:
We cannot be making tens of unnecessary function calls on import for the sake of code readability, so you need to change all of these. I'm happy for you do define default strings, but not to make any function calls in function arguments.

Users don't need to initialise one massdef every time you create a concentration, they only need to initialise whichever single massdef they're gonna use throughout.

Reply:
changed to a string default, but read what I wrote in your other comment